### PR TITLE
Support list for TAAT() PASS_REGULAR_EXPRESSION and FAIL_REGULAR_EXPRESSION (#464)

### PR DESCRIPTION
**Pull request by bartlettroscoe**
See the commit log for commit https://github.com/TriBITSPub/TriBITS/commit/fd681d2823116a813e85f4a39478bc2750011b8e for details.

I noticed this as part of working on testing PR #479 with Trilinos as part of #299.  This should have been addressed as part of PR #464.

### CODE REVIEW

**Review by rabartlett1972 — APPROVED** [review]
@KyleFromKitware, I am going to allow the merge and then a post-merge review can be done with the advantage of having the updated documentation deployed.  Any issues can be addressed with a follow-up review.

**Comment by bartlettroscoe** [comment]
@KyleFromKitware, if you have time, can you please do a post-merge review of this PR?  See the "Review Checklist" [above](https://github.com/TriBITSPub/TriBITS/pull/481#issue-1237806359).  This should not take more than 30 minutes.

**Comment by KyleFromKitware** [comment]
Looks good to me.

File: tribits/core/package_arch/TribitsAddAdvancedTest.cmake [hunk 1]
```diff
@@ -602,35 +602,54 @@ include(PrintVar)
 #
 # **Test case Pass/Fail (tribits_add_advanced_test())**
 #
+# The logic given below can be used to determine pass/fail criteria for a test
+# case both based on what is printed in the test output **and** the return
+# code for the test block command.  Raw CTest, as of version 3.23, does not
+# allow that.  With raw CTest, one can only set determine pass/fail based the
```
Comment (bartlettroscoe) [inline tribits/core/package_arch/TribitsAddAdvancedTest.cmake:608]: "... can set pass/fail criteria based on the test output **or** ..."

**Review by bartlettroscoe — COMMENTED** [review]
I found just a few issues.

File: tribits/core/package_arch/TribitsAddAdvancedTest.cmake [hunk 2]
```diff
@@ -515,14 +515,14 @@ include(PrintVar)
...
-#     given regular expression.  Otherwise, it is assumed to pass.  This will
-#     be applied and take precedence over other above pass criteria.  For
-#     example, if even if ``PASS_REGULAR_EXPRESSION`` or
-#     ``PASS_REGULAR_EXPRESSION_ALL`` match, then the test will be marked as
-#     failed if this fail regex matches the output.
+#     If specified, the test command will be assumed to fail if it matches
+#     **any** of the given regular expressions.  This will be applied and take
+#     precedence over other above pass criteria.  For example, if even if
+#     ``PASS_REGULAR_EXPRESSION`` or ``PASS_REGULAR_EXPRESSION_ALL`` match,
+#     then the test will be marked as failed if this fail regex matches the
```
[the remark this answers is not in the document]
Comment (bartlettroscoe) [reply tribits/core/package_arch/TribitsAddAdvancedTest.cmake:524]: Addressed in #498

File: tribits/core/package_arch/TribitsAddAdvancedTest.cmake (hunk 1 above)
Comment (bartlettroscoe) [reply tribits/core/package_arch/TribitsAddAdvancedTest.cmake:608]: Addressed in #498

File: tribits/core/package_arch/TribitsAddAdvancedTest.cmake [hunk 3]
```diff
@@ -653,15 +679,26 @@ include(PrintVar)
 #
 # However, this can be changed by setting one of the following optional arguments:
 #
-#   ``FINAL_PASS_REGULAR_EXPRESSION <regex>``
+#   ``FINAL_PASS_REGULAR_EXPRESSION "<regex0>" "<regex1>" ...``
 #
 #     If specified, the test will be assumed to pass if the output matches
-#     ``<regex>``.  Otherwise, it will be assumed to fail.
+#     **any** of the provided regular expressions ``<regexi>``.  Otherwise, it
+#     will be assumed to fail.  (Sets the CTest property
```
[the remark this answers is not in the document]
Comment (bartlettroscoe) [reply tribits/core/package_arch/TribitsAddAdvancedTest.cmake:686]: Addressed in #498

**Pull request merged**